lib/board.rb

```diff
@@ -0,0 +1,50 @@
+class Board
+  attr_accessor :cells
+
+  def initialize(cells = Array.new(9, " "))
+    @cells = cells
+  end
+
+  def reset!
+    self.cells = Array.new(9, " ")
+  end
+
+  def display
+    puts row_display(0)
+    puts "------------"
+    puts row_display(3)
+    puts "------------"
+    puts row_display(6)
+  end
+
+  def row_display(i)
+    " #{self.cells[0+i]} | #{self.cells[1+i]} | #{self.cells[2+i]} "
+  end
+
+  def full?
+    self.cells.none? {|c| c == " "}
+  end
+
+  def position(pos)
+    self.cells[pos.to_i-1]
+  end
+
+  def turn_count
+    9 - self.cells.select { |c| c == " "}.count
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```